### PR TITLE
[Eager] Support div(scalar) in eager mode

### PR DESCRIPTION
**Pull request by veyron95**
### PR types
Bug fixes

### PR changes
Others 

### Describe
Support test_tensor_scalar_type_promotion_dynamic.py switch to eager mode.

### CODE REVIEW

File: python/paddle/fluid/dygraph/math_op_patch.py [hunk 1]
```diff
@@ -222,7 +224,7 @@ def __impl__(self, other_var):
                 # so the calculation result here and the calculation result of numpy are 
                 # different after 6 decimal point. If necessary, we can also use float64 here.
                 # torch's behavior here is consistent with ours
-                if op_type == 'elementwise_div' and self.dtype in _supported_int_dtype_:
+                if op_type in _supported_div_op_ and self.dtype in _supported_int_dtype_:
```
Comment (JiabinYang) [inline python/paddle/fluid/dygraph/math_op_patch.py]: this May cause performance issue

**Review by JiabinYang — COMMENTED** [review]
some comments

**Review by JiabinYang — APPROVED** [review]
LGTM

File: python/paddle/fluid/dygraph/math_op_patch.py (hunk 1 above)
Comment (veyron95) [reply python/paddle/fluid/dygraph/math_op_patch.py]: Done, Thx.

**Pull request merged**